src/gc/os.d

```diff
@@ -141,3 +141,54 @@ else
 {
     static assert(false, "No supported allocation methods available.");
 }
+
+/**
+   Check for any kind of memory pressure.
+
+   Params:
+      mapped = the amount of memory mapped by the GC in bytes
+   Returns:
+       true if memory is scarce
+*/
+// TOOD: get virtual mem sizes and current usage from OS
+// TODO: compare current RSS and avail. physical memory
```

Comment:
Just a suggestion, but rather than probing the operating system, why not let the admin/programmer set some ``maxHeap`` parameter which the GC must abide by?